### DISCUSSION TITLE
pre-multiplied alpha when running in overlay mode?

### DISCUSSION THREAD

**Discussion by rmoriz** [time4tea/gopro-dashboard-overlay#203] [Q&A]
Hi,

I'm using ``--generate overlay`` to generate overlay-only clips which I then use in my video editing software of choice (DaVinci Resolve) to combine with the actual footage. I'm using the default coloring option values (no bg options specified, I assume it's `0,0,0,0`) and the following XML layout for the map:

```xml
 <component type="moving_map" name="moving_map" x="3000" y="20" size="800" zoom="17" corner_radius="25" opacity="0.8"/>
```

It works quite well to strip out the background when using the "LumaKeyer" tool but fails when using the simple clip attribute "alpha mode: premultiplied". I assume that using the "pre-multiplied alpha" would render much faster than using the LumaKeyer which takes a long time to render out.

Sadly I see nasty artefacts not only in the map area but also in texts when using "alpha mode: premultiplied":

<img width="418" alt="grafik" src="https://github.com/user-attachments/assets/af975e55-5c97-4c0f-bf99-5b210655cc24">

As far as I understand, Pillow/PIL [has basic support for premultiplied modes](https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes): LA (L with alpha), RGBa (true color with premultiplied alpha. However I've not yet tested it. Maybe I'm completely wrong.

**Comment by williamjoy** [comment]
As a workaround, seems you can create a profile to override the codec, e.g. `png` to preserve alpha channel.

Most of these are from its source code, except adding the `png` one. For command line you can add `--profile png` to override the codec

```json
//cat /root/.gopro-graphics/ffmpeg-profiles.json
{
    "nvgpu": {
        "input": ["-hwaccel", "nvdec"],
        "output": ["-vcodec", "h264_nvenc", "-rc:v", "cbr", "-b:v", "25M", "-bf:v", "3", "-profile:v", "high", "-spatial-aq", "true", "-movflags", "faststart"]
    },
    "nnvgpu": {
        "input": ["-hwaccel", "cuda", "-hwaccel_output_format", "cuda"],
        "filter": "[0:v]scale_cuda=format=yuv420p[mp4_stream];[1:v]format=yuva420p,hwupload[overlay_stream];[mp4_stream][overlay_stream]overlay_cuda",
        "output": ["-vcodec", "h264_nvenc", "-rc:v", "cbr", "-b:v", "25M", "-bf:v", "3", "-profile:v", "main", "-spatial-aq", "true", "-movflags", "faststart"]
    },
    "png": {
        "input": [],
        "output": ["-vcodec", "png"]
    },
    "vp9": {
        "input": [],
        "output": ["-vcodec", "vp9", "-pix_fmt", "yuva420p"]
    },
    "vp8": {
        "input": [],
        "output": ["-vcodec", "vp8", "-pix_fmt", "yuva420p", "-auto-alt-ref", "0"]
    },
    "mac_hevc": {
        "input": ["-hwaccel", "videotoolbox"],
        "output": ["-vcodec", "hevc_videotoolbox", "-q:v", "60"]
    },
    "mac": {
        "input": ["-hwaccel", "videotoolbox"],
        "output": ["-vcodec", "h264_videotoolbox", "-q:v", "60"]
    }
}
```
